### DISCUSSION TITLE
File upload upload files twice

### DISCUSSION THREAD

**Discussion by minhthangdang** [filamentphp/filament#14704] [Help] [answered]
### Package

Form builder

### Package Version

v3

### How can we help you?

Hello,

I'm using the file upload and s3 to store the files. I've realized that when I upload files, it uploads twice, one to a temporary folder when I select files, and another one to the destination folder when I click to save the model. That means it's taking twice the time needed for file uploads. Is there any way I can avoid this? 

Thank you.

**Comment by leandrocfe** [comment, marked as answer]
You mean, [this](https://filamentphp.com/docs/3.x/forms/fields/file-upload#moving-files-instead-of-copying-when-the-form-is-submitted) ?

**Reply by minhthangdang** [reply]
Spot on. Thank you.
